src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinNamesAnnotationIntrospector.kt

```diff
@@ -32,6 +38,10 @@ internal class KotlinNamesAnnotationIntrospector(val module: KotlinModule, val c
                     member.parameterCount == 0) {
                 return member.name.substringAfter("is").decapitalize().substringBefore('-')
             }
+        } else if (member is AnnotatedMethod && member.declaringClass.isKotlinClass()) {
+            if (cache.isKotlinGeneratedMethod(member) { it.declaringClass.declaredFields.any { f -> f.name == member.name } }) {
+                return member.name
```

Comment:
Another possibility would be to implement `com.fasterxml.jackson.databind.introspect.AccessorNamingStrategy` (or maybe extend `DefaultAccessorNamingStrategy` impl) and define `findNameForXxx()`.
But not sure if that would necessarily be any simpler (it could be more efficient as names of all fields, methods could be done just once when creating instance but that's probably not a big deal).